File: myproj26/main07-keyword-arguments.py
def mysum3(x, y, z):
    return x + y * 10 + z * 100


# 위치 인자
print(mysum3(1, 2, 3))

# 키워드 인자
print(mysum3(x=1, y=2, z=3))
print(mysum3(z=3, x=1, y=2))

kwargs = {"x": 1, "y": 2, "z": 3}
mysum3(**kwargs)  # unpacking


people = [
    {"name": "Tom", "age": 10, "region": "Seoul"},
    {"name": "Steve", "age": 12, "region": "Pusan"},
]

# -------------------

for person in people:
    print(person["name"], person["age"])

File: myproj26/main07-keyword-arguments.js
// function mysum3(x, y, z) {
//     return x + y * 10 + z * 100;
// }

// mysum(1, 2, 3)

// js에는 없는 문법
// mysum3(x=1, y=2, z=3)

// 파이썬의 keyword arguments와
// 비슷하게 사용하는 코드

const { x, y, z } = { x: 1, y: 2, z: 3 }

function mysum3({ x, y, z }) {
    return x + y * 10 + z * 100;
}

mysum3({ x: 1, y: 2, z: 3 });

// -------------------

const people = [
    { name: 'Tom', age: 10, region: 'Seoul' },
    { name: 'Steve', age: 12, region: 'Pusan' }
];

for (const { name, age } of people) {
    console.log(name, age);
}